### DISCUSSION TITLE
Weird data gathered using Instrumentation and OTEL

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#67994] [Help]
### Summary

I've enabled instrumentation and I'm using an OTEL provider to gather and visualize the tracing results. I'm seeing some weird results regarding the default Next'js spans regarding the times. It's quite an often issue that skew our data quite a lot and thats why I'm starting this discussion.

Our apps are hosted on Vercel.

If you are using a different OTEL platform, are you seeing similar results?

Could you help me debug some of these issues? Thank you

### Additional information

Here's an example of a trace that goes up to ~4min of execution which the root span accounting for almost 100% of it. This is an API route handler, not a regular page.

<img width="969" alt="image" src="https://github.com/user-attachments/assets/07d42dc3-c39a-4277-b209-b3d974989336">

Similarly, but this is a server action.

<img width="1028" alt="image" src="https://github.com/user-attachments/assets/48262905-47cc-4654-a937-af5b1269dce7">

On another instance, I'm seeing that the whole requests takes quite some time, with the `findPageComponents` span taking almost 50% of that. Why is that? What is this span doing?

<img width="909" alt="image" src="https://github.com/user-attachments/assets/518db95e-2df2-44e3-bd2b-d3ef6e49f670">

And finally, Im getting a report that the `/404` is being hit. When does that url is being hit? 

<img width="373" alt="image" src="https://github.com/user-attachments/assets/067a6edf-a57f-4615-b6d4-8d8914ab28e0">

**Comment by bitttttten** [comment]
I have similar problems and I have 2 answers for only 2 things you are seeing:

1. I think the long handleRequest.next calls are because the lambda is hanging around and not exiting. I think you can look at when `startResponse` ends to get a real sense of when the request ends.
2. Which tier are you using on vercel for your function cpu? https://vercel.com/docs/functions/configuring-functions/memory you could try the performance tier.
